test/Module/Command/CreateTest.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-tooling for the canonical source repository
+ * @copyright Copyright (c) 2017 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-tooling/blob/master/LICENSE.md New BSD License
+ */
+
+namespace ZendTest\Expressive\Tooling\Module\Command;
+
+use org\bovigo\vfs\vfsStream;
+use org\bovigo\vfs\vfsStreamDirectory;
+use phpmock\phpunit\PHPMock;
+use PHPUnit\Framework\TestCase;
+use Zend\Expressive\Tooling\Module\Command\Create;
+use Zend\Expressive\Tooling\Module\Exception;
+
+class CreateTest extends TestCase
+{
+    use PHPMock;
+
+    /** @var vfsStreamDirectory */
+    private $dir;
+
+    /** @var vfsStreamDirectory */
+    private $modulesDir;
+
+    /** @var string */
+    private $composer = 'my-composer';
+
+    /** @var Create */
+    private $command;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->dir = vfsStream::setup('project');
+        $this->modulesDir = vfsStream::newDirectory('my-modules')->at($this->dir);
+        $this->command = new Create($this->dir->url(), 'my-modules', $this->composer);
+    }
+
+    public function testErrorsWhenModuleDirectoryAlreadyExists()
+    {
+        vfsStream::newDirectory('MyApp')->at($this->modulesDir);
+
+        $this->expectException(Exception\RuntimeException::class);
+        $this->expectExceptionMessage('Module "MyApp" already exists');
+        $this->command->process('MyApp');
+    }
+
+    public function testErrorsWhenCannotCreateModuleDirectory()
+    {
+        $baseModulePath = sprintf('%s/my-modules/MyApp', $this->dir->url());
+
+        $mkdir = $this->getFunctionMock('Zend\Expressive\Tooling\Module\Command', 'mkdir');
```

Comment:
Yes, I added:
```json
"php-mock/php-mock-phpunit": "^2.0 || ^1.1.2",
```
There is also integration library with prophecy and mocker.
https://github.com/php-mock
I've chosen integration with PHPUnit as this is the main testing library. I know that many mocking we are doing recently using prophecy, so maybe better choice will be `php-mock-prophecy`?
Here is the library with the main functionality:
https://github.com/php-mock/php-mock